services/localfile/server/localfile.go

```diff
@@ -774,6 +796,50 @@ func (s *server) Symlink(ctx context.Context, req *pb.SymlinkRequest) (*emptypb.
 	return &emptypb.Empty{}, nil
 }
 
+func (s *server) Mkdir(ctx context.Context, req *pb.MkdirRequest) (_ *emptypb.Empty, retErr error) {
+	logger := logr.FromContextOrDiscard(ctx)
+	recorder := metrics.RecorderFromContextOrNoop(ctx)
+
+	if req.GetDirDescription() == nil {
+		recorder.CounterOrLog(ctx, localfileMkdirFailureCounter, 1, attribute.String("reason", "missing_dst"))
+		return nil, status.Error(codes.InvalidArgument, "destination must be filled in")
+	}
+	d := req.GetDirDescription()
+
+	a := d.GetAttrs()
+	if a == nil {
+		recorder.CounterOrLog(ctx, localfileMkdirFailureCounter, 1, attribute.String("reason", "missing_attrs"))
+		return nil, status.Errorf(codes.InvalidArgument, "must send attrs")
+	}
+	filename := a.Filename
+	logger.Info("create directory", filename)
+	dir, _, err := setupOutputDir(a)
+	if err != nil {
+		recorder.CounterOrLog(ctx, localfileMkdirFailureCounter, 1, attribute.String("reason", "setup_output_err"))
+		return nil, err
+	}
+
+	cleanup := func() {
+		if retErr != nil {
+			os.Remove(dir)
+		}
+	}
+	defer cleanup()
+
+	// Do one final check (though racy) to see if the file exists.
```

Comment:
I don't think we need this check. Directory and file renames are a little different and the directory rename should fail if it tries to get renamed to something preexisting.

Reply:
Yes, I check the documentation: https://pkg.go.dev/os#Rename and does will return error if the directory already exist.